cmd/put.go

```diff
@@ -53,68 +55,65 @@ a file to S3.`,
 		if s3object[len(s3object)-1:] == "/" {
 			s3object = s3object + sourceFile
 		}
-		versionId, err := putS3Cse(s3bucket, s3object, viper.GetString("kmskeyid"), sourceFile)
+
+		//Return an encryption client from the global session
+		sess := viper.Get("ClientSess")
+		cmkID := viper.GetString("kmskeyid")
+		newSess := sess.(*session.Session)
+		//Create the KeyProvider
+		handler := s3crypto.NewKMSKeyGenerator(kms.New(newSess), cmkID)
+
+		encryptionclient := Put{
+			Client: clientfactory.NewEncryptyionClient(newSess, s3crypto.AESCBCContentCipherBuilder(handler, crosscrypto.NewPKCS7Padder(16))).S3Client,
+			Bucket: s3bucket,
+			Key:    s3object,
+			Source: sourceFile,
+		}
+
+		versionID, err := encryptionclient.putS3Cse()
 		flagBool := viper.GetBool("verboseoutput")
 
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "err uploading file: %s\n", err)
 			os.Exit(1)
 		}
 		if flagBool {
-			fmt.Fprintf(os.Stdout, "{ \"VersionId\": %s }\n", string(versionId))
+			fmt.Fprintf(os.Stdout, "{ \"versionID\": %s }\n", string(versionID))
```

Comment:
Should we maybe keep the string literal as VersionId? Since that's what end users will be expected - Since it's AWS convention